protocol/x/subaccounts/keeper/isolated_subaccount.go

```diff
@@ -0,0 +1,144 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+// checkIsolatedSubaccountConstaints will validate all `updates` to the relevant subaccounts against
```

Comment:
For my understanding, do we ever expect these constraints to fail during normal operations? If isolated subaccounts is abstracted away from the user, then upstream `x/clob` logic would be responsible for populating the updates correctly right? So this is more of an internal sanity check?

Reply:
Upstream `x/clob` logic relies on  `UpdateSubaccounts` to determine if the updates to a subaccount for a match are valid.  This isn't an internal sanity check, this would be the function / logic that would ensure orders which result in a match that would lead to an invalid subaccount state (based on these new constraints from isolated markets) result in an error.
There is additional non-scoped work to add logic to `x/clob` to prevent users from even sending orders that would lead to matches that are invalid based on the isolated market constraints.